### PR TITLE
Task #164387 chore:Certificate list view changes

### PR DESCRIPTION
**Pull request by tushar33**

### CODE REVIEW

File: src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini [hunk 1]
```diff
@@ -236,3 +236,7 @@ COM_TJCERTIFICATE_SHOW_PUBLIC_PRIVATE_DESC="If 'Private' option is selected then
 COM_TJCERTIFICATE_SHOW_SEARCH_BOX="Show search box on certificate view"
 COM_TJCERTIFICATE_SHOW_SEARCH_BOX_DESC="Select option 'Yes' to show search box"
 
+;Cerificate Menu option
+COM_TJCERTIFICATE_CERTIFICATE_TEXT_FILTER="Show Free Text Filter"
+COM_TJCERTIFICATE_CERTIFICATE_TYPE_FILTER="Show Certificate Type Filter"
+
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini:242]: Remove extra line

File: src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini [hunk 2]
```diff
@@ -52,3 +52,10 @@ COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_FIELD_PRIVATE="Private"
 COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_FIELD_PUBLIC="Public"
 COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_ENTER_CERTIFICATE_ID="Enter Certificate Id"
 
+;__DEPLOY_VERSION__
+;List view
+COM_TJCERTIFICATE_CERTIFICATE_DOWNLOAD="Download"
+COM_TJCERTIFICATE_CERTIFICATE_LIST_VIEW_TYPE="Type"
+COM_TJCERTIFICATE_CERTIFICATE_LIST_VIEW_NAME="Course Name/Event Name"
+COM_TJLMS_COURSE_TJCERT_CLIENT="Course"
+COM_JTICKETING_EVENT_TJCERT_CLIENT="Event"
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini:61]: I know I have told you this format, but this is looking odd wrt to the rest of the lang const. Let's change this to like

`COM_TJCERTIFICATE_CLIENT_COM_TJLMS_COURSE="Course"`
`COM_TJCERTIFICATE_CLIENT_COM_JTICKETING_EVENT="Event"`

File: src/components/com_tjcertificate/site/models/fields/getclientlist.php [hunk 3]
```diff
@@ -59,8 +59,9 @@ protected function getOptions()
 			{
 				foreach ($listobjects as $obj)
 				{
-					$client = explode('.', $obj->client);
-					$options[] = JHtml::_('select.option', $obj->client, ucfirst($client[1]));
+					$client =  str_replace(".", "_", $obj->client);
+					$langConst = strtoupper($client . '_TJCERT_CLIENT');
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/models/fields/getclientlist.php:63]: here also you need changes

File: src/components/com_tjcertificate/site/models/forms/filter_certificates.xml [hunk 4]
```diff
@@ -19,6 +20,7 @@
 			clientByUser="1"
 			onchange="this.form.submit();" />
 
+
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/models/forms/filter_certificates.xml:23]: You can remove this file from the MR as there is no notable changes

File: src/components/com_tjcertificate/site/views/certificates/tmpl/my.php [hunk 5]
```diff
@@ -85,6 +84,21 @@
...
+								{
+									JLoader::import('components.com_tjlms.models.orders', JPATH_SITE);
+									$ordersModel = JModelLegacy::getInstance('Orders', 'TjlmsModel', array('ignore_request' => true));
+									$name = $ordersModel->getCourseName($item->client_id);
+								}
+								elseif ($item->client === "com_jticketing.event")
+								{
+									JLoader::import('components.com_jticketing.helpers.event', JPATH_SITE);
+									$JteventHelper = new JteventHelper;
+									$name = $JteventHelper->getEventData($item->client_id)->title;
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificates/tmpl/my.php:99]: I think here also you can use the same plugin function `getCertificateClientData` to fetch the client details...In this way, any third party extension won't have to hack it

File: src/components/com_tjcertificate/site/views/certificates/tmpl/my.php [hunk 6]
```diff
@@ -106,8 +123,12 @@
 										echo '-';
 									}
 									?></td>
-								<td><?php echo $this->escape($item->comment); ?></td>
-								<td><?php echo (int) $item->id; ?></td>
+								<td>
+									<?php 
+										$client = explode('.', $item->client); 
+										echo ucfirst($client[1]);
+									?>
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificates/tmpl/my.php:130]: You can use lang constant here as you have used in the filters

File: src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini (hunk 1 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini]: done

File: src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini (hunk 2 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini]: done

File: src/components/com_tjcertificate/site/models/fields/getclientlist.php (hunk 3 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/models/fields/getclientlist.php]: done

File: src/components/com_tjcertificate/site/models/forms/filter_certificates.xml (hunk 4 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/models/forms/filter_certificates.xml]: done

File: src/components/com_tjcertificate/site/views/certificates/tmpl/my.php (hunk 5 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificates/tmpl/my.php]: done

File: src/components/com_tjcertificate/site/views/certificates/tmpl/my.php (hunk 6 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificates/tmpl/my.php:120]: done

File: src/components/com_tjcertificate/site/views/certificates/view.html.php [hunk 7]
```diff
@@ -94,6 +94,9 @@ public function display($tpl = null)
 
 		$layout = $app->input->get('layout', "my");
 
+		// Get Menu params
+		$this->menuparams = $this->state->get('parameters.menu');
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificates/view.html.php:98]: Where is this used?

Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificates/view.html.php]: removed

**Pull request closed without merging**